File: server/python/message/message.py
#!/usr/bin/env python
#-*-coding: utf-8-*-

import os, sys
import checker

class Message:
    def __init__(self, holder, post):
        self.__holder = holder
        self.__post = post

    def request(self, data):
        type, err = checker.check_param_type(data)
        if err:
            return err
        if type == 'get_post_count':
            return self.__post.get_count()
        elif type == 'get_all_post':
            return self.__post.get_all()
        elif type == 'get_post_by_id':
            return self.__post.get_by_id(data)
        elif type == 'add_post':
            return self.__post.add(data)
        elif type == 'modify_post':
            return self.__post.modify(data)
        elif type == 'remove_post':
            return self.__post.remove(data)
        else:
            return {'status':'invalid_parameter', 'message':'"type" is invalid.'}

    def close(self):
        self.__holder.close()

File: server/php/message/message.php
<?php

require_once('checker.php');

class Message {
	private $holder;
	private $post;
	
	public function __construct($holder, $post) {
		$this->holder = $holder;
		$this->post = $post;
	}
	
	public function request($data) {
		$typeerr = checkParamType($data);
		$type = $typeerr[0];
		$err = $typeerr[1];
		if ($err) {
			return $err;
		}

		if ($type == 'get_post_count') {
			return $this->post->getCount();
		} else if ($type == 'get_all_post') {
			return $this->post->getAll();
		} else if ($type == 'get_post_by_id') {
			return $this->post->getById($data);
		} else if ($type == 'add_post') {
			return $this->post->add($data);
		} else if ($type == 'modify_post') {
			return $this->post->modify($data);
		} else if ($type == 'remove_post') {
			return $this->post->remove($data);
		} else {
			return ['status'=>'invalid_parameter', 'message'=>'"type" is invalid.'];
		}
	}
	
	public function close() {
		$this->holder->close();
	}
}

?>
